### PR TITLE
Allow not to parse stop_times

### LINKED ISSUE #65: Add an option not to read stop_times
Sometimes we don't need to read the stop_times, It might be worth it to add an option not to read them.

This could speed the GTFS reading and the memory footprint of stuff like [gtfs2geojson](https://gitlab.com/CodeursEnLiberte/gtfs-to-geojson/).

Maybe we can add this only for `RawGTFS` as it might be cleaner to state that the GTFS is not complete?

### PR DESCRIPTION
**Pull request by Tristramg**
I’m not really sure about this approach. It is quite verbose and duplicates a shitton of functions. Could we imagine something template based? Or maybe put the parameters on the basic functions and make a breaking change?

However, like this, we can easilly add a parameter like `trim` 

Closes #65

### CODE REVIEW

**Comment by antoine-de** [comment]
don't you think something like https://github.com/CanalTP/transit_model/pull/780 would be a bit simpler?

TL;DR it's be something like:
```rust
let gtfs = Gtfs::from_path(pouet)?;
```
or
```rust
let gtfs = GtfsReader::default()
               .without_stoptimes()
               .from_path(pouet)?;
```
(and every impl uses under the hood GtfsReader? The question would be how to handle the Gtfs/RawGtfs 🤔

**Comment by Tristramg** [comment]
I like the API you suggest, but I’m not sure how you’d implement it.

An easy solution, is to rename `Parameters` to `GtfsReader`, make all my `_params` methods private, and call them from `GtfsReader`.

Maybe we should turn everything inside-out, and consider that `RawGtfs` is actually the reader

**Comment by Tristramg** [comment]
now shiny with a `GtfsReader`

File: src/raw_gtfs.rs [hunk 1]
```diff
@@ -204,136 +64,56 @@ impl RawGtfs {
...
-        } else {
-            Self::from_path(gtfs)
-        }
+        GtfsReader::default().raw().read(gtfs)
     }
 
     #[cfg(not(feature = "read-url"))]
-    pub fn new(gtfs_source: &str) -> Result<Self, Error> {
-        Self::from_path(gtfs_source)
+    pub fn new(gtfs: &str) -> Result<Self, Error> {
```
Comment (antoine-de) [inline src/raw_gtfs.rs:71]: we don't need this cfg guard anymore

File: src/gtfs_reader.rs [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+            frequencies: read_objs_from_optional_path(&p, "frequencies.txt"),
+            feed_info: read_objs_from_optional_path(&p, "feed_info.txt"),
+            read_duration: Utc::now().signed_duration_since(now).num_milliseconds(),
+            files,
+            sha256: None,
+        })
+    }
+
+    /// Reads from an url (if starts with `"http"`), or a local path (either a directory or zipped file)
+    #[cfg(feature = "read-url")]
```
Comment (antoine-de) [inline src/gtfs_reader.rs:140]: we're missing a non `read-url` feature version of this function

File: src/raw_gtfs.rs [hunk 3]
```diff
@@ -325,7 +365,11 @@ impl RawGtfs {
             calendar_dates: read_optional_file(&file_mapping, &mut archive, "calendar_dates.txt"),
             routes: read_file(&file_mapping, &mut archive, "routes.txt"),
             stops: read_file(&file_mapping, &mut archive, "stops.txt"),
-            stop_times: read_file(&file_mapping, &mut archive, "stop_times.txt"),
+            stop_times: if params.read_stop_times {
+                read_file(&file_mapping, &mut archive, "stop_times.txt")
+            } else {
+                Ok(Vec::new())
```
[the remark this answers is not in the document]
Comment (antoine-de) [reply src/raw_gtfs.rs]: forget about this, we don't have `log` as dependency :stuck_out_tongue_winking_eye:

**Review by antoine-de — APPROVED** [review]

File: Cargo.toml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 [package]
 description = "Read GTFS (public transit timetables) files"
 name = "gtfs-structures"
-version = "0.26.0"
+version = "0.28.0"
```
Comment (antoine-de) [inline Cargo.toml:4]: 0.27.0 is enough no?

Comment (antoine-de) [reply Cargo.toml]: I changed it to 0.27.0

**Review by antoine-de — APPROVED** [review]
